### PR TITLE
Add support for master volume adjustment with key mappings

### PR DESCRIPTION
**Pull request by keatongreve**

### CODE REVIEW

**Comment by FitzRoyX** [comment]
Accidentally hitting a hotkey could cause people to lower their volume without realizing it. Audio changes aren't as easy to spot as graphical ones.

**Comment by keatongreve** [comment]
That's a good point, that was partly why I put the default hotkey behind a modifier key. This is also groundwork for other mechanisms such as native menus or an additional above-game overlay, similar to OpenEmu (I was just looking into this approach).

File: main.c [hunk 1]
```diff
@@ -305,7 +312,8 @@ static void PlayAudio(Snes *snes, SDL_AudioDeviceID device, SDL_AudioFormat form
   if (SDL_GetQueuedAudioSize(device) <= 735 * 4 * 6) {
     int16 *volumeAdjustedAudioBuffer = (int16 *)malloc(735 * 4);
     SDL_memset(volumeAdjustedAudioBuffer, 0, 735 * 4);
-    SDL_MixAudioFormat((Uint8 *)volumeAdjustedAudioBuffer, (Uint8 *)audioBuffer, format, 735 * 4, SDL_MIX_MAXVOLUME);
+    int8 volume_level = (int8)floor((float)g_current_user_volume_level / kMaxUserVolumeLevel * SDL_MIX_MAXVOLUME);
```
Comment (snesrev) [inline main.c:315]: I don't really like floats if it can be avoided.

File: main.c [hunk 2]
```diff
@@ -372,6 +380,12 @@ static void HandleCommand(uint32 j, bool pressed) {
     case kKeys_Turbo: g_turbo = !g_turbo; break;
     case kKeys_ZoomIn: DoZoom(1); break;
     case kKeys_ZoomOut: DoZoom(-1); break;
+    case kKeys_MasterVolumeDown:
+      AdjustMasterVolume(-1 * kVolumeIncrementInterval);
```
Comment (snesrev) [inline main.c:384]: You can write - instead of * -1

**Review by snesrev — COMMENTED** [review]
Is there a way to change volume by notifiying the OS instead (like Windows) - so the modified volume will show up in the Windows volume mixer?

**Comment by keatongreve** [comment]
I think for Windows, we can try [ISimpleAudioVolume](https://docs.microsoft.com/en-us/windows/win32/api/audioclient/nn-audioclient-isimpleaudiovolume) from audioclient.h, but I couldn't determine from documentation if this is the same as the volume mixer level ([This SO post](https://stackoverflow.com/questions/14306048/controlling-volume-mixer) seems to indicate it does.). I'm still looking into it.

**Comment by keatongreve** [comment]
Update: that worked. I need to clean some stuff up and I can send out a PR. just to be sure, how do you want the volume control to be managed going forward? I'm guessing you want to prefer platform-specific mixer APIs unless it is not available (e.g. macOS), in which case we should fall back to the SDL volume mixer?

**Comment by keatongreve** [comment]
also is it okay if the volume adjustment code is a separate library? it's easier to implement in C++ because of COM objects.

**Comment by snesrev** [comment]
Fallback to SDL sounds Okay. I prefer not a separate library. I could translate it to C code if you give me the C++ code for the COM stuff.

**Comment by keatongreve** [comment]
Going to close this for now; integrating the changes from #57 will require a bit more re-working of compilation setups per platform.

**Pull request closed without merging**